lib/elixir/src/elixir_module.erl

```diff
@@ -92,6 +92,10 @@ compile(Line, Module, Block, Vars, E) ->
     RawCompileOpts = bag_lookup_element(DataBag, {accumulate, compile}, 2),
     CompileOpts = validate_compile_opts(RawCompileOpts, AllDefinitions, Unreachable, File, Line),
 
+    OnLoadAttribute = lists:keyfind(on_load, 1, Attributes),
+    OnLoadAttribute /= false andalso
+      validate_on_load_attribute(OnLoadAttribute, AllDefinitions, Unreachable, File, Line),
```

Comment:
I would remove `OnLoadAttribute /= false` and check for false inside `validate_on_load_attribute`.